staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -39,28 +39,45 @@ import (
 )
 
 type validatingDispatcher struct {
-	cm *webhook.ClientManager
+	cm     *webhook.ClientManager
+	plugin *Plugin
 }
 
-func newValidatingDispatcher(cm *webhook.ClientManager) generic.Dispatcher {
-	return &validatingDispatcher{cm}
+func newValidatingDispatcher(p *Plugin) func(cm *webhook.ClientManager) generic.Dispatcher {
+	return func(cm *webhook.ClientManager) generic.Dispatcher {
+		return &validatingDispatcher{cm, p}
+	}
 }
 
 var _ generic.Dispatcher = &validatingDispatcher{}
 
-func (d *validatingDispatcher) Dispatch(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces, relevantHooks []*generic.WebhookInvocation) error {
+func (d *validatingDispatcher) Dispatch(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces, hooks []v1beta1.Webhook) error {
+	var relevantHooks []*generic.WebhookInvocation
 	// Construct all the versions we need to call our webhooks
 	versionedAttrs := map[schema.GroupVersionKind]*generic.VersionedAttributes{}
-	for _, call := range relevantHooks {
+	for i := range hooks {
+		invocation, err := d.plugin.ShouldCallHook(&hooks[i], attr, o)
+		if err != nil {
+			return err
+		}
+		if invocation == nil {
+			continue
+		}
+		relevantHooks = append(relevantHooks, invocation)
 		// If we already have this version, continue
-		if _, ok := versionedAttrs[call.Kind]; ok {
+		if _, ok := versionedAttrs[invocation.Kind]; ok {
 			continue
 		}
-		versionedAttr, err := generic.NewVersionedAttributes(attr, call.Kind, o)
+		versionedAttr, err2 := generic.NewVersionedAttributes(attr, invocation.Kind, o)
 		if err != nil {
-			return apierrors.NewInternalError(err)
+			return apierrors.NewInternalError(err2)
```

Comment:
`if err2 != nil` ? Or can we just use `err` here as well?

Reply:
we can just use err here as well